File: SuperNewRoles/Helpers/DesyncHelpers.cs
using Hazel;
using InnerNet;
using System;
using System.Collections.Generic;
using System.Text;

namespace SuperNewRoles.Helpers
{
    public static class DesyncHelpers
    {
        public static void RPCMurderPlayerPrivate(this PlayerControl source,PlayerControl target,PlayerControl see = null)
        {
            PlayerControl SeePlayer = see;
            if (see == null) SeePlayer = source;
            MessageWriter MurderWriter = AmongUsClient.Instance.StartRpcImmediately(source.NetId, (byte)RpcCalls.MurderPlayer, SendOption.Reliable, SeePlayer.getClientId());
            MessageExtensions.WriteNetObject(MurderWriter, target);
            AmongUsClient.Instance.FinishRpcImmediately(MurderWriter);
        }
        public static void RpcShapeShiftPrivate(this PlayerControl source, PlayerControl target, bool IsAnimation, PlayerControl see = null)
        {
            PlayerControl SeePlayer = see;
            if (see == null) SeePlayer = source;
            MessageWriter val = AmongUsClient.Instance.StartRpcImmediately(source.NetId, 46, (SendOption)1, targetClientId:SeePlayer.getClientId());
            val.WriteNetObject(target);
            val.Write(IsAnimation);
            AmongUsClient.Instance.FinishRpcImmediately(val);
        }
    }
}

File: cpp/Helpers/DesyncHelpers.cpp
#include "DesyncHelpers.h"
#include "../ModHelpers.h"

using namespace Hazel;
using namespace InnerNet;

namespace SuperNewRoles::Helpers
{

	void DesyncHelpers::RPCMurderPlayerPrivate(PlayerControl *source, PlayerControl *target, PlayerControl *see)
	{
		PlayerControl *SeePlayer = see;
		if (see == nullptr)
		{
			SeePlayer = source;
		}
		MessageWriter *MurderWriter = AmongUsClient::Instance->StartRpcImmediately(source->NetId, static_cast<unsigned char>(RpcCalls::MurderPlayer), SendOption::Reliable, SuperNewRoles::ModHelpers::getClientId(SeePlayer));
		MessageExtensions::WriteNetObject(MurderWriter, target);
		AmongUsClient::Instance->FinishRpcImmediately(MurderWriter);
	}

	void DesyncHelpers::RPCMurderPlayerPrivate(PlayerControl *source, CustomRpcSender *sender, PlayerControl *target, PlayerControl *see)
	{
		PlayerControl *SeePlayer = see;
		if (see == nullptr)
		{
			SeePlayer = source;
		}
		sender->StartMessage(SuperNewRoles::ModHelpers::getClientId(SeePlayer))->StartRpc(source->NetId, RpcCalls::MurderPlayer).WriteNetObject(target).EndRpc().EndMessage();
	}
}
